kernel/gic/src/redist_interface.rs

```diff
@@ -0,0 +1,93 @@
+//! Redistributor Interface
```

Comment:
can you describe what the Redistributor is for? Or if it is described elsewhere, link to that here.

Reply:
I added this to dist_interface.rs:
```rust
//! The Distributor forwards or discards SPIs (shared peripheral interrupts)
//! to:
//! - The redistributor, in GICv3,
//! - The CPU core, in GICv2.
//! There's one Distributor per GIC chip.
```
and this to redist_interface.rs:
```rust
//! The Redistributor forwards or discards PPIs (private peripheral interrupts)
//! & SGIs (software generated interrupts) to the CPU core, in GICv3. There's
//! one redistributor per CPU core.
```